### PR TITLE
Bringing this package into the future

### PR DESCRIPTION
**Pull request by jkrems**
* Current npub based release process
* Build against node 0.12 and iojs
* `q` -> `bluebird`
* `npm test` instead of `make test`
* Use mocha's promise support
* Remove `bond`

### CODE REVIEW

File: test/cache.coffee [hunk 1]
```diff
@@ -107,60 +111,84 @@ describe 'Cache', ->
...
+              backend: @cache.backend
+              name: 'awesome-name'
+              debug: true
+            }
+            @failCache.getWrapped = ->
+              Promise.reject new Error('backend get troubles')
+
+          it 'falls back on the value refresher', (done) ->
+            valueGenerator = cached.deferred (cb) ->
+              cb null, 'fresh cats'
```
Comment (Kofia5) [inline test/cache.coffee]: glad to see my data making it through

File: src/cache.coffee [hunk 2]
```diff
@@ -95,16 +96,15 @@ class Cache
...
-
-      Q.npost backend, 'set', [ key, wrappedValue, opts ]
-    ).nodeify cb
+    toPromise(val)
+      .then (resolvedValue) =>
+        @backend.set key, {
+          b: expiresAt(opts.freshFor)
+          d: resolvedValue
+        }, opts
+     .nodeify cb
```
Comment (Kofia5) [inline src/cache.coffee]: spacing not lining up with `.then`

**Comment by Kofia5** [comment]
Generally LGTM but need to test it with an actual app first before signing off

File: src/backends/memcached.coffee [hunk 3]
```diff
@@ -44,19 +45,20 @@ class MemcachedBackend
...
-      if err? then callback(err) else callback(null, value)
+  set: (key, value, options) ->
+    @_clientSet(key, value, options.expire).then -> value
 
-  unset: (key, callback) ->
-    delete @cache[key]
-    callback(null) if callback?
-    null
+  unset: (key) ->
+    @_clientDel(key).then -> undefined
```
Comment (khoomeister) [inline src/backends/memcached.coffee]: Is `then` required in this instance?

Comment (jkrems) [inline src/backends/memcached.coffee]: The `then` makes sure the return value in the success case is undefined. The memcached client doesn't really specify what the return value is.

**Comment by khoomeister** [comment]
Nice work - so much more readable :shipit:

**Pull request merged**